guibot/target.py

```diff
@@ -512,31 +512,62 @@ def load(self, steps_filename, **kwargs):
         :raises: :py:class:`errors.UnsupportedBackendError` if a chain step is of unknown type
         :raises: :py:class:`IOError` if an chain step line cannot be parsed
         """
-        if not os.path.exists(steps_filename):
-            steps_filename = FileResolver().search(steps_filename)
+        def resolve_stepsfile(filename):
```

Comment:
I see you have chosen the nested approach. The private approach is just as valid here and we use both approaches in the project. Let's see what time shows and if we gain more information about pros and cons about these two then. I guess a middle ground could be where a class with many methods and complex implementation makes use of more nesting to simplify the API and a class with fewer methods like this one could tip towards the private method approach.